src/GameConfiguration/index.ts

```diff
@@ -8,38 +8,65 @@ function createHiddenGameConfiguration(revealed: RevealedGameConfiguration, firs
   // expect(firstVisibleState).to.have.property('turn').that.equals(0)
 
   const { grid, tiles } = revealed
-  const visibleMountainTiles = firstVisibleState.tiles.filter(visibleTile => visibleTile.isMountain)
+  const knownMountainTiles = firstVisibleState.tiles.filter(visibleTile => visibleTile.isKnownMountain)
+  const unknownObstacleTiles = firstVisibleState.tiles.filter(visibleTile => visibleTile.isUnknownObstacle)
+
 
   const passable: Set<Tile> = new Set()
+  const unknownObstacles: Set<Tile> = new Set()
   const adjacencies: Map<Tile, Set<Tile>> = new Map()
   const distances: Map<Tile, Map<Tile, number>> = new Map()
+  
+  function isKnownMountain(tile:Tile): boolean {
+    return knownMountainTiles.some(knownMountainTile =>
+      tile.rowIndex === knownMountainTile.rowIndex && tile.colIndex === knownMountainTile.colIndex)
+  }
 
-  function isPassable(tile: Tile): boolean {
-    return !visibleMountainTiles.some(visibleMountain =>
-      tile.rowIndex === visibleMountain.rowIndex && tile.colIndex === visibleMountain.colIndex)
+  function isUnknownObstacle(tile:Tile): boolean {
+    return unknownObstacleTiles.some(unknownObstacleTile =>
+      tile.rowIndex === unknownObstacleTile.rowIndex && tile.colIndex === unknownObstacleTile.colIndex)
+  }
+
+  function isKnownPassable(tile: Tile): boolean {
+    return !isKnownMountain(tile) && !isUnknownObstacle(tile)
   }
 
   function tileAt(rowIndex: number, colIndex: number): Tile | undefined {
     const row = grid[rowIndex]
     return row && row[colIndex]
   }
 
-  tiles.forEach(tile => { if (isPassable(tile)) passable.add(tile) })
+  tiles.forEach(tile => { if (isKnownPassable(tile)) passable.add(tile) })
```

Comment:
I think this line, the line below, and everything having to do with overwriting distances can be eliminated in favor of adjusting line 43 to read like this: 
```typescript
tiles.forEach(from => tiles.forEach(to => distances.get(from)!.set(to, undefined)))
```